test/AsmResolver.DotNet.Tests/Signatures/GenericContextTest.cs

```diff
@@ -72,38 +73,320 @@ public void ResolveMethodGenericParameterWithTypeAndMethod()
             genericType.TypeArguments.Add(_importer.ImportTypeSignature(typeof(string)));
-            
+
             var genericMethod = new GenericInstanceMethodSignature();
             genericMethod.TypeArguments.Add(_importer.ImportTypeSignature(typeof(int)));
-            
+
             var context = new GenericContext(genericType, genericMethod);
-            
+
             var parameter = new GenericParameterSignature(GenericParameterType.Method, 0);
             Assert.Equal("System.Int32", context.GetTypeArgument(parameter).FullName);
         }
 
         [Fact]
-        public void ResolveTypeGenericParameterWithOnlyMethodShouldThrow()
+        public void ResolveTypeGenericParameterWithOnlyMethod()
         {
             var genericInstance = new GenericInstanceMethodSignature();
             genericInstance.TypeArguments.Add(_importer.ImportTypeSignature(typeof(string)));
-            
+
             var context = new GenericContext(null, genericInstance);
-            
+
             var parameter = new GenericParameterSignature(GenericParameterType.Type, 0);
-            Assert.Throws<ArgumentOutOfRangeException>(() => context.GetTypeArgument(parameter));
+            Assert.Equal(parameter,context.GetTypeArgument(parameter));
         }
 
         [Fact]
-        public void ResolveMethodGenericParameterWithOnlyTypeShouldThrow()
+        public void ResolveMethodGenericParameterWithOnlyType()
         {
             var genericInstance = new GenericInstanceTypeSignature(_importer.ImportType(typeof(List<>)), false);
             genericInstance.TypeArguments.Add(_importer.ImportTypeSignature(typeof(string)));
-            
+
             var context = new GenericContext(genericInstance, null);
-            
+
             var parameter = new GenericParameterSignature(GenericParameterType.Method, 0);
-            Assert.Throws<ArgumentOutOfRangeException>(() => context.GetTypeArgument(parameter));
+            Assert.Equal(parameter, context.GetTypeArgument(parameter));
+        }
+
+
+        [Fact]
+        public void ParseGenericFromTypeSignature()
+        {
+            var genericInstance = new GenericInstanceTypeSignature(_importer.ImportType(typeof(List<>)), false);
+            genericInstance.TypeArguments.Add(_importer.ImportTypeSignature(typeof(string)));
+
+            var context = GenericContext.FromType(genericInstance);
+            var context2 = GenericContext.FromMember(genericInstance);
+            var context3 = GenericContext.FromType((ITypeDescriptor)genericInstance);
+
+            Assert.Equal(context, context3);
+            Assert.Equal(context, context2);
+            Assert.False(context.IsEmpty);
+            Assert.Equal(genericInstance, context.Type);
+            Assert.Null(context.Method);
+        }
+
+        [Fact]
+        public void ParseGenericFromNonGenericTypeSignature()
+        {
+            var type = new TypeDefinition("","Test type", TypeAttributes.Public);
```

Comment:
Missing spaces in format. Few other places in this same file with the same problem.